netty/src/main/java/io/grpc/transport/netty/NettyClientStream.java

```diff
@@ -124,7 +133,7 @@ protected void sendFrame(WritableBuffer frame, boolean endOfStream, boolean flus
       bytebuf = ((NettyWritableBuffer) frame).bytebuf();
     }
     channel.write(new SendGrpcFrameCommand(this, bytebuf, endOfStream));
-    if (flush) {
+    if (flush && !channel.eventLoop().inEventLoop()) {
```

Comment:
I'm uncertain, but we may possibly be in the same event loop, but for a different Channel. Like if we are servicing a request and make an outgoing call.